src/text/stylized.rs

```diff
@@ -0,0 +1,224 @@
+use super::Span;
+use crate::style::{Color, Modifier, Style};
+
+pub trait Styled {
+    type Item;
+
+    fn style(&self) -> Style;
+    fn set_style(self, style: Style) -> Self::Item;
+}
+// Workaround for https://github.com/rust-lang/rust/issues/78835
+macro_rules! calculated_docs {
+    ($(#[doc = $doc:expr] $item:item)*) => { $(#[doc = $doc] $item)* };
+}
+
+// https://github.com/crossterm-rs/crossterm/blob/1efdce7ef63cba6992729db5f22262a60936fa8b/src/style/stylize.rs#L5-L56
+macro_rules! stylize_method {
+    ($method_name:ident Modifier::$modifier:ident) => {
```

Comment:
It could be worth using https://github.com/dtolnay/paste#case-conversion and https://github.com/dtolnay/paste#pasting-documentation-strings to simplify this to e.g.:
```rust
color_method!(red);
...
modifier_method(bold);
...
```

(Not a blocker, just small simplification of the code)